packages/llvm.rb

```diff
@@ -1,26 +1,283 @@
+  description 'The LLVM Project (version 6.0.0) is a collection of modular and reusable compiler and toolchain technologies. The optional packages including clang, lld, polly, compiler-rt, libcxx, libcxxabi and openmp are included.'
   homepage 'http://llvm.org/'
-  version '6.0.0'
-    source_url 'http://llvm.org/releases/6.0.0/llvm-6.0.0.src.tar.xz'
+  version '6.0.0-1'
+  source_url 'https://llvm.org/releases/6.0.0/llvm-6.0.0.src.tar.xz'
   source_sha256 '1ff53c915b4e761ef400b803f07261ade637b0c269d99569f18040f3dcee4408'
 
   binary_url ({
-    aarch64: 'https://dl.bintray.com/chromebrew/chromebrew/llvm-6.0.0-chromeos-armv7l.tar.xz',
-     armv7l: 'https://dl.bintray.com/chromebrew/chromebrew/llvm-6.0.0-chromeos-armv7l.tar.xz',
-       i686: 'https://dl.bintray.com/chromebrew/chromebrew/llvm-6.0.0-chromeos-i686.tar.xz',
-     x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/llvm-6.0.0-chromeos-x86_64.tar.xz',
+    aarch64: 'https://dl.bintray.com/chromebrew/chromebrew/llvm-6.0.0-1-chromeos-armv7l.tar.xz',
+     armv7l: 'https://dl.bintray.com/chromebrew/chromebrew/llvm-6.0.0-1-chromeos-armv7l.tar.xz',
+       i686: 'https://dl.bintray.com/chromebrew/chromebrew/llvm-6.0.0-1-chromeos-i686.tar.xz',
+     x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/llvm-6.0.0-1-chromeos-x86_64.tar.xz',
   })
   binary_sha256 ({
-    aarch64: '9bfffb161e2e3c0642178175fe7ce2c18168c8ab71bce91e10ad4a15b80e3ea7',
-     armv7l: '9bfffb161e2e3c0642178175fe7ce2c18168c8ab71bce91e10ad4a15b80e3ea7',
-       i686: '4e3c4ee05546a808fadfbd7f388ce1f64e8daac72c1eaa7e396ed062c0c94328',
-     x86_64: 'e24785612a16133605fb93fb9fd43d663b3a1e69e6cb69e5c518f23dab8321a3',
+    aarch64: '4c770487aeaa3c86ae1b3154ca4155111454171b1f82416120f73beb8f0e4ded',
+     armv7l: '4c770487aeaa3c86ae1b3154ca4155111454171b1f82416120f73beb8f0e4ded',
+       i686: 'a2c64fd6d83c91e5ecfaebef5a95e54c6843f9370f4d2ba6b936ca61a3ec205a',
+     x86_64: '99267bc8144bc6359eac2fc9810f0c8c814eadcad06e5d167a542d1639c7f9c0',
   })
 
-  depends_on 'cmake' => :build
+  depends_on 'python27' => :build # for test suite
+  depends_on 'icu4c'              # fix uconv.h missing
+  depends_on 'libtirpc'
+  
+  
+  def self.preinstall
+    
+    ############################# Download clang (tools) ###########################################
+    url_clang = 'https://llvm.org/releases/6.0.0/cfe-6.0.0.src.tar.xz'
+    uri_clang = URI.parse url_clang
+    filename_clang = File.basename(uri_clang.path)
+    sha256sum_clang = 'e07d6dd8d9ef196cfc8e8bb131cbd6a2ed0b1caf1715f9d05b0f0eeaddb6df32'
+    
+    if File.exist?(filename_clang) && Digest::SHA256.hexdigest( File.read("./#{filename_clang}") ) == sha256sum_clang
+      puts "Clang source code exists and perform unpacking".yellow
+      system "tar", "xf", "cfe-6.0.0.src.tar.xz", "-C", "tools"
+      puts "Clang source code unpacked".lightgreen
+      system "rm -rf tools/clang"  # remove possible existing folder
+      system "mv -v tools/cfe-6.0.0.src tools/clang"
```

Comment:
You could move lines 38-41 outside the if statement since lines 49-51 duplicate the process.